code/datums/components/crafting/crafting.dm

```diff
@@ -119,24 +125,33 @@
 	.["tool_behaviour"] = list()
 	.["other"] = list()
 	.["instances"] = list()
-	for(var/obj/item/I in get_environment(a,blacklist))
-		if(.["instances"][I.type])
-			.["instances"][I.type] += I
-		else
-			.["instances"][I.type] = list(I)
-		if(istype(I, /obj/item/stack))
-			var/obj/item/stack/S = I
-			.["other"][I.type] += S.amount
-		else if(I.tool_behaviour)
-			.["tool_behaviour"] += I.tool_behaviour
-			.["other"][I.type] += 1
-		else
-			if(istype(I, /obj/item/reagent_containers))
-				var/obj/item/reagent_containers/RC = I
-				if(RC.is_drainable())
-					for(var/datum/reagent/A in RC.reagents.reagent_list)
-						.["other"][A.type] += A.volume
-			.["other"][I.type] += 1
+	.["machinery"] = list()
+	for(var/obj/I in get_environment(a,blacklist))
+		if(istype(I, /obj/item))
+			var/obj/item/item = I
+			if(.["instances"][item.type])
+				.["instances"][item.type] += item
+			else
+				.["instances"][item.type] = list(item)
+			if(istype(item, /obj/item/stack))
+				var/obj/item/stack/S = item
+				.["other"][item.type] += S.amount
+			else if(item.tool_behaviour)
+				.["tool_behaviour"] += item.tool_behaviour
+				.["other"][item.type] += 1
+			else
+				if(istype(item, /obj/item/reagent_containers))
+					var/obj/item/reagent_containers/RC = item
+					if(RC.is_drainable())
+						for(var/datum/reagent/A in RC.reagents.reagent_list)
+							.["other"][A.type] += A.volume
+				.["other"][I.type] += 1
+		else if (istype(I, /obj/machinery))
+			if(.["machinery"][I.type])
+				.["machinery"][I.type] += I
+			else
+				.["machinery"][I.type] = list(I)
```

Comment:
```suggestion
	for(var/obj/object in get_environment(a, blacklist))
		if(isitem(object))
			var/obj/item/item = object
			LAZYADDASSOC(.["instances"], [item.type], item)
			if(isstack(item))
				var/obj/item/stack/stack = item
				.["other"][item.type] += stack.amount
			else if(item.tool_behaviour)
				.["tool_behaviour"] += item.tool_behaviour
				.["other"][item.type] += 1
			else
				if(istype(item, /obj/item/reagent_containers))
					var/obj/item/reagent_containers/container = item
					if(RC.is_drainable())
						for(var/datum/reagent/reagent in container.reagents.reagent_list)
							.["other"][reagent.type] += reagent.volume
				.["other"][I.type] += 1
		else if (ismachinery(object))
			LAZYADDASSOC(.["machinery"], [object.type], object)
```
Pretty sure this works fine, but it's good to test changes, and even more web-edits.